cpp/src/DataStorm/NodeSessionI.h

```diff
@@ -16,38 +17,34 @@ namespace DataStormI
     class NodeSessionI : public std::enable_shared_from_this<NodeSessionI>
     {
     public:
-        NodeSessionI(
-            std::shared_ptr<Instance>,
-            std::shared_ptr<DataStormContract::NodePrx>,
-            std::shared_ptr<Ice::Connection>,
-            bool);
+        NodeSessionI(std::shared_ptr<Instance>, std::optional<DataStormContract::NodePrx>, Ice::ConnectionPtr, bool);
```

Comment:
I wouldn't mind having some comments; in particular, explain what it means for instance/node/connection to be null or nullopt.